plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go

```diff
@@ -626,3 +627,180 @@ func copyTestChartsIntoHarness(t *testing.T, th *kusttest_test.HarnessEnhanced)
 }
+
+func TestHelmChartInflationGeneratorWithSameChartMultipleVersions(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t).
+		PrepBuiltin("HelmChartInflationGenerator")
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+	tests := []struct {
+		name        string
+		chartName   string
+		repo        string
+		version     string
+		releaseName string
+	}{
+		{
+			name:        "terraform chart with no version grabs latest",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "",
+			releaseName: "terraform-latest",
+		},
+		{
+			name:        "terraform chart with version 1.1.1",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "1.1.1",
+			releaseName: "terraform-1.1.1",
+		},
+		{
+			name:        "terraform chart with version 1.1.1 again",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "1.1.1",
+			releaseName: "terraform-1.1.1-1",
+		},
+		{
+			name:        "terraform chart with version 1.1.2",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "1.1.2",
+			releaseName: "terraform-1.1.2",
+		},
+		{
+			name:        "terraform chart with version 1.1.2 but without repo",
+			chartName:   "terraform",
+			version:     "1.1.2",
+			releaseName: "terraform-1.1.2",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := fmt.Sprintf(`
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+ name: %s
+name: %s
+version: %s
+repo: %s
+releaseName: %s
+`, tt.chartName, tt.chartName, tt.version, tt.repo, tt.releaseName)
+
+			rm := th.LoadAndRunGenerator(config)
+			assert.True(t, len(rm.Resources()) > 0)
+
+			var chartDir string
+			if tt.version != "" && tt.repo != "" {
+				chartDir = fmt.Sprintf("charts/%s-%s/%s", tt.chartName, tt.version, tt.chartName)
+			} else {
+				chartDir = fmt.Sprintf("charts/%s", tt.chartName)
+			}
+
+			fmt.Printf("%s: %s\n", tt.name, chartDir)
+
+			d, err := th.GetFSys().ReadFile(filepath.Join(th.GetRoot(), chartDir, "Chart.yaml"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			assert.Contains(t, string(d), fmt.Sprintf("name: %s", tt.chartName))
+			if tt.version != "" {
+				assert.Contains(t, string(d), fmt.Sprintf("version: %s", tt.version))
+			}
+		})
+	}
+}
+
+// Test that verifies +1 instances of same chart with different versions
+// https://github.com/kubernetes-sigs/kustomize/issues/4813
+func TestHelmChartInflationGeneratorWithMultipleInstancesSameChartDifferentVersions(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t).
+		PrepBuiltin("HelmChartInflationGenerator")
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	podinfo1 := th.LoadAndRunGenerator(`
```

Comment:
I think it looks like this test case called the generator two times.
If my understanding is wrong, please point it out.

Reply:
This is expected since it simulates the same chart with a different version.
Helm Repo: https://stefanprodan.github.io/podinfo
Chart Name: podinfo
Version: 6.1.8 and 6.2.1